csrc/multidevice/utils.h

```diff
@@ -65,15 +65,20 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
+// Returns a set that contains DIDs and Stream.
+std::unordered_set<ParallelType> deviceAndStreamParallelTypes();
+
+// Returns all parallel types that can appear during multi-GPU scheduling, i.e.
+// DIDs, Stream, and Serial.
+std::unordered_set<ParallelType> allParallelTypes();
```

Comment:
The name seems misleading here.
Can we rename to `allMultideviceParallelTypes` or something indicating this is for multi-GPU parallel types.

Reply:
In my latest commit, I took a slightly different solution. I made allParallelTypes return really all parallel types and use that with shardAllLike. Because shardAllLike is called before segmentation, it should only see and propagate multi-GPU parallel types, i.e., DIDs, Stream and Serial.